rust/protocol/src/storage/traits.rs

```diff
@@ -1,38 +1,59 @@
 //
-// Copyright 2020 Signal Messenger, LLC.
+// Copyright 2020-2021 Signal Messenger, LLC.
 // SPDX-License-Identifier: AGPL-3.0-only
 //
 
+//! Traits defining several stores used throughout the Signal Protocol.
+
 use async_trait::async_trait;
 use uuid::Uuid;
 
-use crate::state::{PreKeyId, SignedPreKeyId};
 use crate::{
-    IdentityKey, IdentityKeyPair, PreKeyRecord, ProtocolAddress, Result, SenderKeyRecord,
-    SessionRecord, SignedPreKeyRecord,
+    address::ProtocolAddress,
+    error::Result,
+    sender_keys::SenderKeyRecord,
+    state::{PreKeyId, PreKeyRecord, SessionRecord, SignedPreKeyId, SignedPreKeyRecord},
+    IdentityKey, IdentityKeyPair,
 };
 
+/// Handle to FFI-provided context object.
+///
+/// This object is not manipulated in rust, but is instead passed back to each FFI
+/// method invocation.
 pub type Context = Option<*mut std::ffi::c_void>;
 
+/// Each Signal message can be considered to have exactly two participants, a sender and receiver.
+///
+/// [IdentityKeyStore::is_trusted_identity] uses this to ensure the identity provided is configured
+/// for the appropriate role.
 #[derive(Debug, Clone, Eq, PartialEq)]
 pub enum Direction {
     Sending,
     Receiving,
 }
 
+/// Interface defining the TOFU identity store, which may be in-memory, on-disk, etc.
 #[async_trait(?Send)]
 pub trait IdentityKeyStore {
+    /// Return the single specific identity each store is assumed to represent, with private key.
     async fn get_identity_key_pair(&self, ctx: Context) -> Result<IdentityKeyPair>;
 
+    /// Return a [u32] specific to this store instance.
+    ///
+    /// This numerical id should be randomly generated each time the store is instantiated.
```

Comment:
This is not correct; the local registration ID is a per-device identifier (what you've called "client instance" elsewhere) and should not change run to run.

Reply:
Thanks so much for this clear feedback! I will make this change!

Question:
> (what you've called "client instance" elsewhere)

Is that referring to the changes in #286? I have a small followup PR from your comments at https://github.com/signalapp/libsignal-client/pull/286/files#r625973774 which makes `DeviceId` into its own struct and I will make sure to stitch this into there as well.

Reply:
Somewhat yeah. Within Signal "a device" usually refers to a particular installation that's been registered with a particular user and assigned a device ID (thus being uniquely identified by a ProtocolAddress), but the term you picked, "client instance", might be more accessible.

As I commented in your follow-up PR, not all of these are device IDs. :-)